src/test/scala/com/github/vitalsoftware/util/RobustPrimitivesTest.scala

```diff
@@ -0,0 +1,25 @@
+package com.github.vitalsoftware.util
+
+import com.github.vitalsoftware.macros.jsonDefaults
+import org.specs2.mutable.Specification
+import org.specs2.specification.Scope
+import play.api.libs.json._
+
+@jsonDefaults case class Primitives(f1: Int = 1, f2: String = "2", f3: Boolean = true, f4: Option[Test])
+
+class RobustPrimitivesTest extends Specification {
+
+  trait TestScope extends RobustPrimitives with Scope
+
+  "RobustPrimitives" should {
+    "read as string" in new TestScope {
+      Reads.of[String].reads(JsBoolean(true)) mustEqual (JsSuccess("true"))
+      Reads.of[String].reads(JsBoolean(false)) mustEqual (JsSuccess("false"))
+      Reads.of[String].reads(JsNumber(123)) mustEqual (JsSuccess("123"))
+      Reads.of[String].reads(JsNumber(123.456)) mustEqual (JsSuccess("123.456"))
+
```

Comment:
Can you add a failure case in there as well?
I've noticed that `BigDecimal(string)` isn't wrapped in a `Try`. Should it be? Should you produce a `JsError` when an expected number is instead of "1234" something like ">15%" or "I'm actually just a string"?

Reply:
Oops, haven't completed this yet. Forgot to add the WIP. I'll be careful with the `BigDecimal`. Thanks